### ISSUE TITLE
Timeline Editor -- animating: Elements, values, transformations by KeyFrames 

### ISSUE THREAD

**Issue by weekendkoder** [PhaserEditor2D/PhaserEditor#102]
I would like to see a **Timeline Editor** inside PhaserEditor2D v2.xx. The great benefit would be:
- animating values, by Keyframes
- own individual easings, by Keyframes
- animating elements, by Keyframes
- animating transformations, by Keyframes
etc.

(We allready have a Sprite-Animator-Editor. What i mean is a multi-purpose Timeline-Animation-Editor)

Really endless possibilities ! A great way to control your animations, beside the coding method. We know this allready from other Apps / GameEngines. It has its reasons why these Apps / GameEngines have it on board. So it would be great, if we got such a Tool inside PhaserEditor2d v2.xx. I am sure, the Devs have this on their track :-)

**Comment by PhaserEditor2D** [comment]
Yes, this is something we want to do before other bigger features. I will think when it is the right moment to do it. There are other stuff like Matter physics, Tilemap, and prefabs.

**Comment by PhaserEditor2D** [comment]
Also, we should think if we are going to implement it as a Scene tool or a separate tool, with a custom file format, that you can add to your own asset libs and use in different projects.

**Comment by weekendkoder** [comment]
Yep, first things first. Good to know, its on your **to-do-list**. 

I would prefer a "scene tool", so integrated into PhaserEditor. Otherwise we have "seperate" :
- tilemap editor
- sprite editor
- code editor
- etc.

Than, where is the benefit of such a "Phaser3" - Editor. Just my opinion ... ;-)

**Comment by PhaserEditor2D** [comment]
Yes, it is in our todo list, you can check the development Project, in the Propositions column, a card about it: https://github.com/PhaserEditor2D/PhaserEditor/projects/1#card-1272973

I do not understand very well your concern about "separate" tools. Maybe I did not explain well. You can see, we have an Animations JSON editor. The Animations JSON is a format from Phaser 3, that we implemented in the IDE. It is a separate editor inside the Phaser Editor IDE, but you can link the animations in the scene editor. In Phaser Editor v1, you create the animations inside the scene editor, what is handly but is more difficult to reuse in other games or in other scenes (unless you create a prefab).

If we create a timeline format and a sepparated editor inside the IDE (like the asset pack editor, the animations editor), then it is more reusable between scenes and games. You can design characters that can be packed and used in different scenarios (including the scene editor, that will be very integrated with the timeline editor).

However, we know there are lot of use cases where you may need a timeline inside the scene editor, to animate different entities at the same time, animate menu screens, etc..., not only a single entity and its parts. So we have to think how to do this, there is nothing decided yet. The "natural" solution is a timeline inside the scene, probably this is the one we are going to implement first.

**Comment by weekendkoder** [comment]
Aha ok. Maybe i misinterpret it with "separate". I am really new to PhaserEditor. I like the way how the SpriteAnimator Module is integrated inside PhaserEditor, from what i saw in videos. Something similiar i could imagine to happen with such a "TimeLine Editor", or i wish to happen.  

> ... Animations JSON ... is a separate editor inside the Phaser Editor IDE ...

"**inside** the Phaser Editor IDE" , is the part that i like ! With "separate" i thought you meant a separate Program, a seperate executable ... something outside PhaserEditor. My fault :-)

**Comment by PhaserEditor2D** [comment]
No, no, nothing is going to be separated, we are building a huge IDE :)
